Source/Core/Core.swift

```diff
@@ -481,11 +481,36 @@ public class FormViewController : UIViewController, FormViewControllerProtocol {
     
     public override func viewWillAppear(animated: Bool) {
         super.viewWillAppear(animated)
-        if let selectedIndexPath = tableView?.indexPathForSelectedRow {
-            tableView?.reloadRowsAtIndexPaths([selectedIndexPath], withRowAnimation: .None)
-            tableView?.selectRowAtIndexPath(selectedIndexPath, animated: false, scrollPosition: .None)
-            tableView?.deselectRowAtIndexPath(selectedIndexPath, animated: true)
+
+        let selectedIndexPaths = tableView?.indexPathsForSelectedRows ?? []
+        tableView?.reloadRowsAtIndexPaths(selectedIndexPaths, withRowAnimation: .None)
+        selectedIndexPaths.forEach {
+            tableView?.selectRowAtIndexPath($0, animated: false, scrollPosition: .None)
+        }
+
+        let deselectionAnimation = { (context: UIViewControllerTransitionCoordinatorContext) in
```

Comment:
@mtnbarreto no need to `[weak self]`, since you're in an animation block that `self` is not retaining. Only need `[weak self]` when the thing that the block is capturing (and therefore retaining) is also retaining the block. Here, the animation system is holding the block, and will eventually release it, which will then release the captured `self`.